mrtt-ui/src/hooks/useAuth.js

```diff
@@ -0,0 +1,40 @@
+import { createContext, useContext, useMemo } from 'react'
+import { useLocalStorage } from './useLocalStorage'
+import PropTypes from 'prop-types'
+import { isExpired } from 'react-jwt'
+import { useState } from 'react'
+
+const AuthContext = createContext()
+
+export const AuthProvider = ({ children }) => {
+  const [token, setToken] = useLocalStorage('token', null)
```

Comment:
if I am reading this right, we only use `useLocalStorage` here. Maybe better just to use the js way of accessing that and not make a react abstraction for it?

Reply:
Same reason as above. I probably just don't have a full grasp of React yet so feel free to re-implement as needed.